src/wireguard.rs

```diff
@@ -0,0 +1,305 @@
+
+use crate::Result;
+
+
+// Some constants
+
+const SIZE_CK: usize = 32;
+const SIZE_HS: usize = 32;
+const SIZE_PUBKEY: usize = 32;
+const SIZE_PRIVKEY: usize = 32;
+
+
+#[allow(missing_docs)]
+// core wireguard app
+pub struct Wireguard <T : trussed::Client>
+{
+    trussed: T,
+}
+
+/*
+    Commands / Requests
+ */
+
+ /*
+  get AEAD                        params: pubkey, C, H     - The most important function, since the AEAD is necessary to perform the NOISE handshake.
+                                                              - Transmitts the peer's public key, the chained key C and chained Hash 'H' to the device. Returns AEAD.
+                                                              - Requires: User action (e.g. Button interaction) on first use. Perhaps additionaly every X minutes (configurable)
+ */
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct GetAead 
+ {
+     pub pubkey: [u8;SIZE_PUBKEY],
+     pub c : [u8; SIZE_CK],
+     pub h : [u8; SIZE_HS],
+     pub key_id : u32
+ }
+
+ //  Authenticate                    params: PIN/pass         - Unlocks the device to use it. 
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct Unlock
+ {
+     pub pin: String, // pin code to unlock the device
+ }
+
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct RegisterKeyPair
+ {
+     pub privkey : [u8; SIZE_PRIVKEY],
+     pub pubkey : [u8; SIZE_PUBKEY],
+     pub label : String
+ }
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct UpdateKeyPair
+ {
+    pub privkey : [u8; SIZE_PRIVKEY],
+    pub pubkey : [u8; SIZE_PUBKEY],
+    pub label : String
+ }
+
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct DeleteKeyPair
+ {
+     pub uid : u32 // unique ID 
+ }
+
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct GenerateKeyPair
+ {
+    pub uid : u32, // unique ID 
+    pub label : String
+ }
+ 
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct ListKeys
+ {
+    //empty
+ }
+ 
+
+
+ /*
+    Reponses
+ */
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct AEAD(pub [u8;32]);
+
+
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct KeyResponse
+ {
+    pub key_info : KeyInfo
+ }
+
+
+ /**/
+
+ // To be serialized and safed in the trussed store
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct KeyInfo 
+ {
+
+    id : u64,
+    label : String,
+    pubkey : [u8; SIZE_PUBKEY]
+ }
+
+
+// implementations
+
+// hex
+impl core::fmt::Display for AEAD {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{:x?}", self.0)
+    }
+}
+
+impl core::fmt::Display for KeyResponse {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{:x?} w/ UID: {} pubkey : {:x?}", self.key_info.id, self.key_info.label, self.key_info.pubkey)
```

Comment:
Again, seems more `Debug` than `Display`. In any case, might be useful to look at https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.debug_struct if you're not aware.